src/robot/htmldata/template.py

```diff
@@ -14,16 +14,29 @@
 #  limitations under the License.
 
 import os
-from os.path import abspath, dirname, join, normpath
+import pathlib
+import sys
 
+if sys.version_info < (3, 10) and not os.path.exists(__file__):
+    # Try importlib resources backport as prior to python 3.10
+    # importlib.resources.files was not zipapp compatible...
+    try:
+        from importlib_resources import files
+    except ImportError:
+        raise ImportError("robotframework outside of filesystem (zipapp?) requires importlib resources backport on python < 3.10")
+else:
+    try:
+        from importlib.resources import files
+    except ImportError:
+        # python 3.8 or earlier:
+        def files(modulepath):
+            base_dir = pathlib.Path(__file__).parent.parent.parent
+            return base_dir / modulepath.replace(".", os.sep)
 
-class HtmlTemplate:
-    _base_dir = join(dirname(abspath(__file__)), '..', 'htmldata')
-
-    def __init__(self, filename):
-        self._path = normpath(join(self._base_dir, filename.replace('/', os.sep)))
-
-    def __iter__(self):
-        with open(self._path, encoding='UTF-8') as file:
-            for line in file:
-                yield line.rstrip()
+def HtmlTemplate(filename):
+    module, filename = os.path.split(os.path.normpath(filename))
+    module = 'robot.htmldata.' + module
+    
+    with files(module).joinpath(filename).open('r', encoding="utf-8") as f:
+        for item in f:
+            yield item.rstrip()
```

Comment:
This looks good otherwise, but I still don't understand what was the reason to change this from a class to a function. In the current usage there's no difference, but having a function named in `CamelCase` is always strange. The function could obviously be renamed, but a class works here as well and allows further configuration in the future if needed. Unless you have concrete reasons for the change, please change this back to a class.